processingchain/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/processingchain/Workflow.groovy

```diff
@@ -7,6 +7,155 @@ import org.orbisgis.datamanager.JdbcDataSource
+
+            if(inputFolder){
+                def outputDir = new File(outputFolder)
+                def folder = new File(inputFolder)
+                if(folder.isDirectory()){
+                    def shapeFiles  = []
+                    folder.traverse(type: groovy.io.FileType.FILES, nameFilter: ~/.*\.shp/) { File shapeFile ->
+                        shapeFiles << shapeFile.getAbsolutePath()
+                    }
+                    if(!outputFolder){
+                        outputDir=new File("$inputFolder${File.separator}results")
+                        if(!outputDir.exists()){
+                            outputDir.mkdir()
+                        }
+                    }
+                    else {
+                        if(!outputDir.isDirectory()){
+                            error "Please set a valid output folder"
+                            return null
+                        }
+                    }
+                    //Looking for IRIS_GE shape file
+                    if(shapeFiles.find{ it.toLowerCase().endsWith("iris_ge.shp")}) {
+                        //Load the shapefiles
+                        shapeFiles.each { shp ->
+                            datasource.load(shp, true)
+                        }
+                        //Prepare iteration on each insee code
+                        def inseeCodes = []
+                        datasource.eachRow("select distinct insee_com from IRIS_GE group by insee_com ;") { row ->
+                            inseeCodes << row.insee_com
+                        }
+
+                        //Let's run the BDTopo process for each insee code
+                        def prepareBDTopoData = ProcessingChain.PrepareBDTopo.prepareBDTopo()
+                        int nbAreas = inseeCodes.size();
+                        info "$nbAreas areas will be processed"
+                        inseeCodes.eachWithIndex { code, index->
+                         if(prepareBDTopoData.execute([datasource                 : datasource,
+                                                    tableIrisName              : 'IRIS_GE', tableBuildIndifName: 'BATI_INDIFFERENCIE',
+                                                    tableBuildIndusName        : 'BATI_INDUSTRIEL', tableBuildRemarqName: 'BATI_REMARQUABLE',
+                                                    tableRoadName              : 'ROUTE', tableRailName: 'TRONCON_VOIE_FERREE',
+                                                    tableHydroName             : 'SURFACE_EAU', tableVegetName: 'ZONE_VEGETATION',
+                                                    tableImperviousSportName   : 'TERRAIN_SPORT', tableImperviousBuildSurfName: 'CONSTRUCTION_SURFACIQUE',
+                                                    tableImperviousRoadSurfName: 'SURFACE_ROUTE', tableImperviousActivSurfName: 'SURFACE_ACTIVITE',
+                                                    distBuffer                 : 500, expand: 1000, idZone: code,
```

Comment:
concerning all parameters having default values, I wonder if we really need to keep them in the processing chain or workflow IProcess when they are not defined as input of these workflow or processing chains. If we modify the default value distBuffer in the prepareBDTopoData IProcess for example, we will have to modify the value of the parameter for all IProcesses that call prepareBDTopo if we want to keep the same default value everywhere.

Reply:
Call conf needed

Reply:
Done extract same parameters